final-project/scripts/main.js

```diff
@@ -1,10 +1,148 @@
 import "../styles/main.scss";
+import "../styles/css-loader/scss/index.scss";
 
-// testing that `const` and arrow functions transpile correctly
-const before = "Hello -world!";
-const after = before
-  .split("")
-  .filter(char => char !== "-")
-  .join("");
-console.log({ before });
-console.log({ after });
+const apiUrl = 'https://wt-4662f45b9eefda7172b747b28d23efdb-0.sandbox.auth0-extend.com/blog';
+let articles;
+let slideIndicators = [];
+let currentArticleIndex = 0;
+
+function bookmark(title, url) {
+  if(document.all) { // ie
+    window.external.AddFavorite(url, title);
+  }
+  else if(window.sidebar) { // firefox
+    window.sidebar.addPanel(title, url, "");
+  }
+  else if(window.opera && window.print) { // opera
+    let elem = document.createElement('a');
+    elem.setAttribute('href', url);
+    elem.setAttribute('title', title);
+    elem.setAttribute('rel','sidebar');
+    elem.click(); // this.title=document.title;
+  } else {
+    // For the other browsers (mainly WebKit) we use a simple alert to inform users that they can add to bookmarks with ctrl+D/cmd+D
+    alert('You can add this page to your bookmarks by pressing ' + (navigator.userAgent.toLowerCase().indexOf('mac') != - 1 ? 'Command/Cmd' : 'CTRL') + ' + D on your keyboard.');
+
+  }
+}
+
+const setLoader = ({visible}) => {
+  const loaderContainer = document.querySelector('.blog .loader-container');
+  const mainSlide = document.getElementById('main_slide');
+  if (visible) {
+    loaderContainer.classList.remove("hidden");
+    mainSlide.classList.add("hidden");
+  } else{
+    loaderContainer.classList.add('hidden');
+    mainSlide.classList.remove('hidden');
+  }
+};
+
+const getImages = (article) => {
+  const pictureContainer = document.createElement('picture');
+  const viewPorts = [
+    {
+      viewport: 'desktop',
+      mediaQuery: '(min-width: 1024px)',
+    },
+    {
+      viewport: 'tablet',
+      mediaQuery: '(min-width: 768px)',
+    },
+    {
+      viewport: 'mobile',
+      mediaQuery: '(min-width: 481px)',
+    },
+    {
+      viewport: 'mobile2x',
+      mediaQuery: '(min-width: 320px)',
+    },
+  ];
+  viewPorts.forEach((viewPort) => {
+    const source = document.createElement('source');
+    source.srcset = article.images[viewPort.viewport];
+    source.media = viewPort.mediaQuery;
+    pictureContainer.appendChild(source);
+  });
+  const defaultImage = document.createElement('img');
+  defaultImage.src = article.images.desktop;
+  pictureContainer.appendChild(defaultImage);
+  return pictureContainer;
+};
+
+const renderArticle = (article) => {
+  setLoader({visible: false});
+  const title = document.querySelector('.blog .slide__title');
```

Comment:
Just as a recommendation, when selecting nodes with js is better to do it with selectors that are not related with styles, ie: `js-blog-slide__title`